addons/mail/static/src/messaging/entity/messaging_initializer/messaging_initializer.js

```diff
@@ -21,56 +21,35 @@ function MessagingInitializerFactory({ Entity }) {
          * the current user. This includes pinned channels for instance.
          */
         async start() {
-            await this.constructor._start(this);
-        }
-
-        /**
-         * Called when messaging is stopped.
-         */
-        stop() {
-            if (this.messaging.device) {
-                this.messaging.device.stop();
-            }
-        }
-
-        //----------------------------------------------------------------------
-        // Private
-        //----------------------------------------------------------------------
-
-        /**
-         * Static method in order to be patched.
-         *
-         * @static
-         * @private
-         * @param {mail.messaging.entity.MessagingInitializer} messagingInitializer
-         */
-        static async _start(messagingInitializer) {
             await this.env.session.is_bound;
 
             this.env.entities.Thread.create({
                 id: 'inbox',
+                isPinned: true,
                 model: 'mail.box',
                 name: this.env._t("Inbox"),
             });
             this.env.entities.Thread.create({
                 id: 'starred',
+                isPinned: true,
                 model: 'mail.box',
                 name: this.env._t("Starred"),
             });
             this.env.entities.Thread.create({
                 id: 'history',
+                isPinned: true,
                 model: 'mail.box',
                 name: this.env._t("History"),
             });
 
-            messagingInitializer.messaging.link({
-                attachmentViewer: this.env.entities.AttachmentViewer.create(),
-                chatWindowManager: this.env.entities.ChatWindowManager.create(),
-                device: this.env.entities.Device.create(),
-                dialogManager: this.env.entities.DialogManager.create(),
-                discuss: this.env.entities.Discuss.create(),
-                locale: this.env.entities.Locale.create(),
-                messagingMenu: this.env.entities.MessagingMenu.create(),
+            this.messaging.update({
+                attachmentViewer: [['create']],
+                chatWindowManager: [['create']],
+                device: [['create']],
+                dialogManager: [['create']],
+                discuss: [['create']],
+                locale: [['create']],
+                messagingMenu: [['create']],
```

Comment:
Those entities need messaging, so I would "inverse the link" (basically, give messaging as param to them on their create) which probably implies to explicitly define the inverse relation (and that relation should likely be "required" and "readonly" if those are parameters that we support).

Reply:
Follow up on this remark: does "create" command actually make sense? There are probably not many use cases, and maybe others can be replaced similarly?